From cd2f67c39b25de026390bfe2bc1c7aa269f78ccb Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Thu, 10 Jan 2019 16:03:06 +0000
Subject: Fix elapsed time check

Summary:
To match the code it was refactored from it should be checking the code
since the last time we checked calcSpeed, not since the job started.

CCBUG: 402665

Subscribers: kde-frameworks-devel

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D18158
---
 src/core/slaveinterface.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/core/slaveinterface.cpp b/src/core/slaveinterface.cpp
index 6eb2b08..d79b498 100644
--- a/src/core/slaveinterface.cpp
+++ b/src/core/slaveinterface.cpp
@@ -103,7 +103,9 @@ void SlaveInterface::calcSpeed()
     // using first and last item from the list.
 
     const qint64 elapsed_time = d->elapsed_timer.elapsed();
-    if (elapsed_time >= 900) {
+    const qint64 last_time = d->transfer_details.isEmpty() ? 0 : d->transfer_details.last().time;
+
+    if (elapsed_time - last_time >= 900) {
         if (d->transfer_details.count() == max_count) {
             d->transfer_details.removeFirst();
         }
-- 
cgit v1.1

